src/test/java/org/kainos/ea/serviceTests/JobRoleServiceTest.java

```diff
@@ -0,0 +1,47 @@
+package org.kainos.ea.serviceTests;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.kainos.ea.dao.JobRoleDao;
+import org.kainos.ea.exceptions.DatabaseConnectionException;
+import org.kainos.ea.exceptions.FailedToGetJobRoleException;
+import org.kainos.ea.model.JobRoleRequest;
+import org.kainos.ea.service.JobRoleService;
+import org.kainos.ea.utility.DatabaseConnector;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class JobRoleServiceTest {
```

Comment:
I think we are missing a scenario here for when the database connector throws an exception?

Reply:
We've added this now.